File: src/main/java/java2kotlin/application/dogs/InMemoryDogRepository.java
package java2kotlin.application.dogs;

import java2kotlin.domain.dog.Dog;
import java2kotlin.domain.dog.IDogRepository;
import org.springframework.stereotype.Component;

import java.util.*;

@Component
public class InMemoryDogRepository implements IDogRepository {

    private Map<String, Dog> dogs = new HashMap<>();

    @Override
    public List<Dog> all() {
        return new ArrayList<>(dogs.values());
    }

    @Override
    public Dog findById(String id) {
        return dogs.get(id);
    }

    @Override
    public Dog save(Dog dog) {
        final String id;

        if(dog.getId() == null){
            id = UUID.randomUUID().toString();
        }
        else {
            id = dog.getId();
        }

        dog.setId(id);
        dogs.put(id, dog);
        return dog;
    }
}

File: src/main/java/java2kotlin/application/dogs/InMemoryDogRepository.kt
package java2kotlin.application.dogs

import java2kotlin.domain.dog.Dog
import java2kotlin.domain.dog.IDogRepository
import org.springframework.stereotype.Component
import java.util.HashMap
import java.util.UUID

@Component
class InMemoryDogRepository : IDogRepository {
    private val dogs: MutableMap<String, Dog> = HashMap()

    override fun all(): List<Dog> = dogs.values.toList()

    override fun findById(id: String): Dog? = dogs[id]

    override fun save(dog: Dog): Dog {
        val id: String = if (dog.id == null) {
            UUID.randomUUID().toString()
        } else {
            dog.id
        }
        dog.id = id
        dogs[id] = dog
        return dog
    }
}
